Source/RapyutaSimulationPlugins/Public/Robots/RRBaseRobot.h

```diff
@@ -556,73 +620,23 @@ class RAPYUTASIMULATIONPLUGINS_API ARRBaseRobot : public ARRBaseActor
     UFUNCTION(BlueprintCallable)
     virtual bool InitPropertiesFromJSON();
 
-    // UI WIDGET --
-    UPROPERTY()
-    uint8 bUIWidgetEnabled : 1;
-    //! UI widget component
-    UPROPERTY()
-    TObjectPtr<UWidgetComponent> UIWidgetComp = nullptr;
-    //! #URRUserWidget's widget
-    UPROPERTY()
-    TObjectPtr<URRUserWidget> UIUserWidget = nullptr;
-
-    //! Relative pose of the UI widget from the owner robot
-    UPROPERTY()
-    FTransform UIWidgetOffset = FTransform(FVector(0.f, 0.f, 100.f));
-
-    /**
-     * @brief Check whether #UIUserWidget is valid
-     */
-    bool CheckUIUserWidget() const;
-
-    /**
-     * @brief Set robot's tooltip text through #UIWidgetComp's label
-     * @param InTooltip
-     */
-    void SetTooltipText(const FString& InTooltip);
-    /**
-     * @brief Toggle robot's tooltip visibility
-     * @param bInTooltipVisible
-     */
-    void SetTooltipVisible(bool bInTooltipVisible);
-
     /**
-     * @brief Set visibility of #UIUserWidget
-     * @param bInWidgetVisible
-     */
-    void SetUIWidgetVisible(bool bInWidgetVisible);
-
-protected:
-    /**
-     * @brief Instantiate default child components
-     */
-    virtual void PreInitializeComponents() override;
-
-    /**
-     * @brief Post Initialization process of actor. Initialize #RobotVehicleMoveComponent by calling #InitMoveComponent.
-     * @sa[ActorLifecycle](https://docs.unrealengine.com/5.1/en-US/ProgrammingAndScripting/ProgrammingWithCPP/UnrealArchitecture/Actors/ActorLifecycle/)
-     * @sa[PostInitializeComponents](https://docs.unrealengine.com/5.1/en-US/API/Runtime/Engine/GameFramework/AActor/PostInitializeComponents/)
-     */
-    virtual void PostInitializeComponents() override;
-
-    /**
-     * @brief Create and Initialize #MovementComponent if #VehicleMoveComponentClass != nullptr.
-     * If VehicleMoveComponentClass == nullptr, it is expected that MovementComponent is set from BP or user code.
+     * @brief This method is called inside #PreInitializeComponents.
+     * Custom initialization of child BP class can be done by overwritting this method.
      *
-     * @return true #MovementComponent is created and initialized.
-     * @return false #VehicleMoveComponentClass == nullptr.
      */
-    virtual bool InitMoveComponent();
+    UFUNCTION(BlueprintImplementableEvent, BlueprintCallable)
+    void BPInitPropertiesFromJSON();
 
     /**
-     * @brief This method is called inside #PostInitializeComponents.
-     * Custom initialization of child class can be done by overwritting this method.
+     * @brief Calls both #InitPropertiesFromJSON and #BPInitPropertiesFromJSON
      *
      */
-    virtual void ConfigureMovementComponent();
-
-    /**
-     * @brief Create & init #UIWidgetComp
-     */
-    virtual void InitUIWidget();
+    UFUNCTION(BlueprintCallable)
+    virtual bool InitPropertiesFromJSONAll()
+    {
+        bool res = InitPropertiesFromJSON();
+        BPInitPropertiesFromJSON();
+        return res;

```

Comment:
```suggestion
        return InitPropertiesFromJSON() && BPInitPropertiesFromJSON();
```
if the BP method could return bool